src/unistd.rs

```diff
@@ -447,7 +447,14 @@ pub fn daemon(nochdir: bool, noclose: bool) -> Result<()> {
     Errno::result(res).map(drop)
 }
 
-pub fn sethostname(name: &[u8]) -> Result<()> {
+/// Set the system host name (see
+/// [gethostname(2)](http://man7.org/linux/man-pages/man2/gethostname.2.html)).
+///
+/// Given a name, attempt to update the system host name to the given string.
+/// On some systems, the host name is limited to as few as 64 bytes.  An error
+/// will be return if the name is not valid or the current process does not have
+/// permissions to update the host name.
+pub fn sethostname(name: &str) -> Result<()> {
```

Comment:
> &str can contain a NUL byte anywhere, and is UTF-8, which may contain "special characters". Not really sure what to do here though!

That's kind of where I landed as well.  Thinking about this more, I'm wondering what you think about the following:

```rust
pub fn sethostname<S: AsRef<OsStr>>(name: S) -> Result<()> {
```

This would allow a greater variety of types (including strings) to work with the function and communicates a little more clearly that we are basically going to let the OS do validation.